pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -426,14 +426,36 @@ func (cmvt *cdcMixedVersionTester) initWorkload(
 
 func (cmvt *cdcMixedVersionTester) muxRangeFeedSupported(
 	r *rand.Rand, h *mixedversion.Helper,
-) (bool, error) {
+) (bool, option.NodeListOption, error) {
+	// The mux setting was added in 22.2 and deleted in 24.1, so we need to first
+	// determine if the cluster version is in the range [22.2, 24.1).
 	cv, err := h.ClusterVersion(r)
 	if err != nil {
-		return false, err
+		return false, nil, err
+	}
+	if !cv.AtLeast(roachpb.MustParseVersion(v222CV)) ||
+		cv.AtLeast(roachpb.MustParseVersion(v241CV)) {
+		return false, nil, nil
+	}
+
+	// Since the setting was deleted in 24.1, there may be nodes running binaries
+	// that no longer know about the setting.
+
+	// If we are upgrading to a version < 24.1, the setting is known to all nodes.
+	if h.Context.ToVersion.Major() < 24 {
+		return true, h.Context.CockroachNodes, nil
 	}
-	// Mux setting only exists in 22.2 to 23.2 inclusive.
-	return cv.AtLeast(roachpb.MustParseVersion(v222CV)) &&
-		!cv.AtLeast(roachpb.MustParseVersion(v241CV)), nil
+
+	// If we are upgrading to 24.1, the setting is only known to pre-24.1 nodes.
+	if h.Context.ToVersion.Major() == 24 && h.Context.ToVersion.Minor() == 1 {
+		nodesInPreviousVersion := h.Context.NodesInPreviousVersion()
+		if len(nodesInPreviousVersion) > 0 {
+			return true, nodesInPreviousVersion, nil
```

Comment:
A potential way to simplify, or just for awareness: this could also be written as:

```go
if h.Context.ToVersion.Major() == 24 && h.Context.ToVersion.Minor() == 1 && h.Context.MixedBinary() {
    return true, h.Context.NodesInPreviousVersion(), nil
}
```

Reply:
Updated, thanks for the tip!